packages/experimental/Menu/src/MenuItem/useMenuItem.ts

```diff
@@ -51,12 +58,21 @@ export const useMenuItem = (props: MenuItemProps): MenuItemState => {
   );
 
   const pressable = useAsPressable({ ...rest, disabled, onPress: onInvoke });
+  const itemRef = useViewCommandFocus(componentRef);
   const keys = isSubmenu ? submenuTriggerKeys : triggerKeys;
 
   // Explicitly override onKeyDown to override the native behavior of moving focus with arrow keys.
   const onKeyDownProps = useKeyDownProps(onInvoke, ...keys);
   const hasCheckmarks = useMenuListContext().hasCheckmarks;
 
+  React.useEffect(() => {
+    if (pressable.state.hovered) {
+      componentRef?.current?.focus();
+    } else {
+      componentRef?.current?.blur();
+    }
+  }, [pressable.state.hovered, componentRef]);
```

Comment:
Do we also need this for MenuItemCheckBox / MenuItemRadio?

Reply:
Yeah, we do. Thanks for catching.

Reply:
Is this still an issue? I wasn't sure if your last refactor meant this is shared now

Reply:
Ah the refactoring only merged the types, I didn't touch any of the render impl since they're somewhat different between the two components.

Reply:
Anyway thanks for catching, added the behavior to MenuItemCheckbox (and Radio) too.